core/src/avm1/activation.rs

```diff
@@ -575,9 +575,9 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     }
 
     fn action_add(&mut self) -> Result<FrameControl<'gc>, Error<'gc>> {
-        let a = self.context.avm1.pop();
-        let b = self.context.avm1.pop();
-        let result = b.into_number_v1() + a.into_number_v1();
+        let a = self.context.avm1.pop().coerce_to_f64(self)?;
```

Comment:
`b` should be coerced before `a`:
```as
var a = { valueOf: function() { trace("a"); } };
var b = { valueOf: function() { trace("b"); } };
trace(a + b); // traces "b", then "a"
```
This is also true for at least `Add2`, and probably others, which in Ruffle the order is wrong.

Reply:
Good catch, thank you! I fixed `Add`, as well as `Less` - `Equals` has a different behavior. I can do additional testings in the next days to fix `Add2` and check the other actions, unless you already have plans for this.